File: day02/snd.jl
open("input.txt") do f
  input = read(f, String) |> strip
  global rs = split(input, ",") .|> (x -> split(x, "-") .|> x -> parse(Int, x))
end

check = x -> occursin(r"^(\d+)\1+$", string(x))
checkr = (a, b) -> [i for i=a:b + 1 if check(i)]

println([e for r in rs for e in checkr(r[1], r[2])] |> sum)

File: day08/snd.py
from operator import itemgetter, sub
from itertools import combinations
from networkx.utils.union_find import UnionFind

with open("input.txt") as f:
    cs = map(
        lambda line: tuple(map(int, line.split(","))), f.read().strip().split("\n")
    )
ds = lambda a, b: sum(map(lambda x: x**2, map(lambda z: sub(*z), zip(a, b))))
dists = {ds(a, b): (a, b) for a, b in combinations(cs, 2)}
uf = UnionFind()
last = (0, 0)
for d, (a, b) in sorted(dists.items(), key=itemgetter(0)):
    if uf[a] == uf[b]:
        continue
    uf.union(a, b)
    last = (a[0], b[0])
print(last[0] * last[1])
